tool/teleport/common/debug.go

```diff
@@ -0,0 +1,255 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"path/filepath"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/debug"
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+)
+
+// debugServiceClient debug service client.
+type debugServiceClient struct {
+	clt *http.Client
+}
+
+// newDebugServiceClient generates a new debug service client.
+func newDebugServiceClient(configPath string) (*debugServiceClient, error) {
+	cfg, err := config.ReadConfigFile(configPath)
```

Comment:
Updated it. If the config is empty, we return to the default data dir path. It seems to be a sufficient handling, given that if the data dir is incorrect, the requests will fail accordingly.

Reply:
I'll update the error message to make it clearer.

Also, should we provide a flag for users to specify their data directory directly? When they get this error, it might be easier to set the data directory directly instead of providing the path to their Teleport config.